lib/ansible/modules/cloud/docker/docker_swarm.py

```diff
@@ -0,0 +1,519 @@
+            - Maximum number of tasks history stored.
+            - Docker default value is C(5).
+    snapshot_interval:
+        description:
+            - Number of logs entries between snapshot.
+            - Docker default value is C(10000).
+    keep_old_snapshots:
+        description:
+            - Number of snapshots to keep beyond the current snapshot.
+            - Docker default value is C(0).
+    log_entries_for_slow_followers:
+        description:
+            - Number of log entries to keep around to sync up slow followers after a snapshot is created.
+    heartbeat_tick:
+        description:
+            - Amount of ticks (in seconds) between each heartbeat.
+            - Docker default value is C(1s).
+    election_tick:
+        description:
+            - Amount of ticks (in seconds) needed without a leader to trigger a new election.
+            - Docker default value is C(10s).
+    dispatcher_heartbeat_period:
+        description:
+            - The delay for an agent to send a heartbeat to the dispatcher.
+            - Docker default value is C(5s).
+    node_cert_expiry:
+        description:
+            - Automatic expiry for nodes certificates.
+            - Docker default value is C(3months).
+    name:
+        description:
+            - The name of the swarm.
+    labels:
+        description:
+            - User-defined key/value metadata.
+    signing_ca_cert:
+        description:
+            - The desired signing CA certificate for all swarm node TLS leaf certificates, in PEM format.
+    signing_ca_key:
+        description:
+            - The desired signing CA key for all swarm node TLS leaf certificates, in PEM format.
+    ca_force_rotate:
+        description:
+            - An integer whose purpose is to force swarm to generate a new signing CA certificate and key,
+                if none have been specified.
+            - Docker default value is C(0).
+    autolock_managers:
+        description:
+            - If set, generate a key and use it to lock data stored on the managers.
+            - Docker default value is C(no).
+        type: bool
+    rotate_worker_token:
+        description: Rotate the worker join token.
+        type: bool
+        default: 'no'
+    rotate_manager_token:
+        description: Rotate the manager join token.
+        type: bool
+        default: 'no'
+extends_documentation_fragment:
+    - docker
+requirements:
+    - python >= 2.7
+    - Docker API >= 1.35
+author:
+  - Thierry Bouvet (@tbouvet)
+'''
+
+EXAMPLES = '''
+
+- name: Init a new swarm with default parameters
+  docker_swarm:
+    state: present
+    advertise_addr: 192.168.1.1
+
+- name: Update swarm configuration
+  docker_swarm:
+    state: present
+    election_tick: 5
+
+- name: Add nodes
+  docker_swarm:
+    state: join
+    advertise_addr: 192.168.1.2
+    join_token: SWMTKN-1--xxxxx
+    remote_addrs: [ '192.168.1.1:2377' ]
+
+- name: Leave swarm
+  docker_swarm:
+    state: absent
```

Comment:
This one I still find confusing. I would expect this would remove/destroy the swarm ?
What is leaving the swarm in this case ? (It may just be terminology and I simply am that ignorant guy)

Reply:
- absent state: remove/destroy a 'all' swarm. 
- remove state: if a node leave a swarm, it is not removed from the swarm configuration, it can re-connect to the manager later. So remove state is to purge the swarm configuration, and after, node can not connect again.

Reply:
Ok, so the description in that example does not match the action, correct ?

Reply:
Problem is that if it's one host, this node leaves the swarm cluster and if the host is all the cluster, the cluster will be destroyed.